src/main/java/pl/pelikan/service/UserService.java

```diff
@@ -0,0 +1,80 @@
+package pl.pelikan.service;
+
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import pl.pelikan.model.User;
+import pl.pelikan.model.UserRole;
+import pl.pelikan.repository.UserRepository;
+import pl.pelikan.repository.UserRoleRepository;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class UserService {
+
+    private static final String DEFAULT_ROLE = "ROLE_USER";
+    private final UserRepository userRepository;
+    private final UserRoleRepository roleRepository;
+    private final PasswordEncoder passwordEncoder;
+
+    public UserService(UserRepository userRepository, UserRoleRepository roleRepository, PasswordEncoder passwordEncoder) {
+        this.userRepository = userRepository;
+        this.roleRepository = roleRepository;
+        this.passwordEncoder = passwordEncoder;
+    }
+
+    public void registerUser(User user) {
+        UserRole defaultRole = roleRepository.findByRole(DEFAULT_ROLE);
+        user.getRoles().add(defaultRole);
+        String passwordHash = passwordEncoder.encode(user.getPassword());
+        user.setPassword(passwordHash);
+        userRepository.save(user);
+    }
+
+    public User findByUsername(String username) {
+        return userRepository.findByUsername(username).get();
+    }
+
+    public void updateUser(User user) {
+        userRepository.save(user);
+    }
+
+    public void changeCredentials(User user) {
+        User toUpdate = findCurrentUser();
```

Comment:
Nie lepiej zrobić tylko setPassword na użytkowniku jaki mamy w parametrze? Wtedy jak dojdzie nam nowe pole do usera, to nie będziemy musieli pamiętać, że tu kolejnego seta musimy dopisać